### ISSUE TITLE
add support for tracking speech markers in bidirectional streams

### ISSUE THREAD

**Issue by davehorton** [jambonz/jambonz-feature-server#811]
We should have some feature like [twilio's mark event](https://www.twilio.com/docs/voice/media-streams/websocket-messages#send-a-mark-message) for bidrectional streams, to allow the remote party to track roughly which pieces of speech have been played.  The idea is that while sending bidirectional streaming audio, they can intermittently -- perhaps at the end of each sentence for example, send a mark message and then we send a play_done message back when the audio to that point has been played out.

**Comment by vykintazo** [comment]
@davehorton are marks in `listen` verb functional or not, since they are already documented as being so: https://docs.jambonz.org/verbs/verbs/listen#mark ?

However I have been encountering issues and behaviour which suggests marks are being sent out almost immediately Jambonz receives them.

I am trying to sync the audio chunks I am sending at network/CPU speed by interleaving the marks: `chunk | mark | chunk | mark | ...` (this simplified test scenario to reproduce the issue) but if I send N chunks and marks this way in a single millisecond, I later receive all N marks again at the same time, while I expect them to be spaced out by `chunk` duration. And now I find this open issue suggesting this hasn't been implemented yet? Or did I misunderstand how `marks` work? As it seems `mod_audio_fork` is no longer open source, I have no way to verify this.

What I am trying to achieve in general is to provide Jambonz with some buffer to counter jitter while still being able to determine when realtime playout ends.

**Comment by davehorton** [comment]
Yes they are functional, and have been used by several customers.  If you are having issues open a support ticket with us by sending details to support@jambonz.org
